TCAT/Controllers/RouteDetail+DrawerViewController.swift

```diff
@@ -252,87 +210,166 @@ UIGestureRecognizerDelegate, PulleyDrawerViewControllerDelegate {
+    func expandCells(on cell: UITableViewCell) {
+        if justLoaded { summaryTapped() }
 
-        // Create Footer for No Data from Live Tracking Footer, if needed
+        toggleCellExpansion(at: tableView.indexPath(for: cell))
 
-        guard
-            let drawer = self.parent as? RouteDetailViewController,
-            let contentViewController = drawer.primaryContentViewController as? RouteDetailContentViewController
-            else {
-                return emptyFooterView
-        }
+        tableView.layoutIfNeeded()
+        tableView.layoutSubviews()
+    }
+}
 
-        var message: String?
+extension RouteDetailDrawerViewController: PulleyDrawerViewControllerDelegate {
+    func collapsedDrawerHeight(bottomSafeArea: CGFloat) -> CGFloat {
+        return bottomSafeArea + summaryView.frame.height
+    }
 
-        if !contentViewController.noDataRouteList.isEmpty {
-            if contentViewController.noDataRouteList.count > 1 {
-                message = Constants.Banner.noLiveTrackingForRoutes
+    func partialRevealDrawerHeight(bottomSafeArea: CGFloat) -> CGFloat {
+        return main.height / 2
+    }
+
+    func drawerPositionDidChange(drawer: PulleyViewController, bottomSafeArea: CGFloat) {
+
+        justLoaded = false
+        // Center map on drawer change
+        switch drawer.drawerPosition {
+        case .collapsed, .partiallyRevealed:
+            guard let contentViewController = drawer.primaryContentViewController as? RouteDetailContentViewController
+                else { return }
+            if let direction = selectedDirection {
+                if direction.type == .walk {
+                    contentViewController.centerMap(on: direction, isOverviewOfPath: true)
+                } else {
+                    contentViewController.centerMap(on: direction)
+                }
+                selectedDirection = nil
             } else {
-                let routeNumber = contentViewController.noDataRouteList.first!
-                message = Constants.Banner.noLiveTrackingForRoute + " " + "\(routeNumber)."
+                contentViewController.centerMapOnOverview(drawerPreviewing: drawer.drawerPosition == .partiallyRevealed)
             }
-        } else {
-            message = nil
+        default: break
         }
+    }
 
-        if let message = message {
-            let phraseLabelFooterView = tableView.dequeueReusableHeaderFooterView(withIdentifier: Constants.Footers.phraseLabelFooterView)
-                as? PhraseLabelFooterView ?? PhraseLabelFooterView(reuseIdentifier: Constants.Footers.phraseLabelFooterView)
-            phraseLabelFooterView.setView(with: message)
-            return phraseLabelFooterView
+    func drawerChangedDistanceFromBottom(drawer: PulleyViewController, distance: CGFloat, bottomSafeArea: CGFloat) {
+
+        // Manage cover view hiding drawer when collapsed
+        if distance - bottomSafeArea == summaryView.frame.height {
```

Comment:
can you assign `distance - bottomSafeArea == summaryView.frame.height` to a boolean so that its clear what it represents